usecases/config/runtime/manager_test.go

```diff
@@ -0,0 +1,397 @@
+		# HELP weaviate_runtime_config_last_load_success Whether the last loading attempt of runtime config was success
+		# TYPE weaviate_runtime_config_last_load_success gauge
+		weaviate_runtime_config_last_load_success 0
+		`)))
+	})
+
+	t.Run("valid config should succeed creating config manager at the startup", func(t *testing.T) {
+		reg := prometheus.NewPedanticRegistry()
+
+		tmp, err := os.CreateTemp("", "valid_config.yaml")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, os.Remove(tmp.Name()))
+		})
+
+		buf := []byte(`backup_interval: 10s`)
+
+		_, err = tmp.Write(buf) // valid yaml
+		require.NoError(t, err)
+		require.NoError(t, tmp.Close())
+
+		_, err = NewConfigManager(tmp.Name(), parseYaml, 10*time.Millisecond, log, reg)
+		require.NoError(t, err)
+
+		// assert: config_last_load_success=1 and config_hash should be set.
+		assert.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(fmt.Sprintf(`
+                # HELP weaviate_runtime_config_hash Hash value of the currently active runtime configuration
+        	# TYPE weaviate_runtime_config_hash gauge
+        	weaviate_runtime_config_hash{sha256="%s"} 1
+		# HELP weaviate_runtime_config_last_load_success Whether the last loading attempt of runtime config was success
+		# TYPE weaviate_runtime_config_last_load_success gauge
+		weaviate_runtime_config_last_load_success 1
+		`, fmt.Sprintf("%x", sha256.Sum256(buf))))))
+	})
+
+	t.Run("changing config file should reload the config", func(t *testing.T) {
+		reg := prometheus.NewPedanticRegistry()
+
+		// calling parser => reloading the config
+		loadCount := 0
+		trackedParser := func(buf []byte) (*testConfig, error) {
+			loadCount++
+			return parseYaml(buf)
+		}
+
+		tmp, err := os.CreateTemp("", "valid_config.yaml")
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, os.Remove(tmp.Name()))
+		})
+
+		buf := []byte(`backup_interval: 10s`)
+
+		_, err = tmp.Write(buf) // valid yaml
+		require.NoError(t, err)
+		require.NoError(t, tmp.Close())
+
+		cm, err := NewConfigManager(tmp.Name(), trackedParser, 10*time.Millisecond, log, reg)
+		require.NoError(t, err)
+
+		// assert: should have called `parser` only once during initial loading.
+		assert.Equal(t, 1, loadCount)
+
+		// Now let's change the config file few times
+		var (
+			wg          sync.WaitGroup
+			ctx, cancel = context.WithCancel(context.Background())
+		)
+		defer cancel() // being good citizen.
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			cm.Run(ctx)
+		}()
+
+		n := 3 // number of times we change the config file after initial reload
+		writeDelay := 100 * time.Millisecond
+		for i := 0; i < n; i++ {
+			// write different config every time
+			buf := []byte(fmt.Sprintf("backup_interval: %ds", i+1))
+			err := os.WriteFile(tmp.Name(), buf, 0o777)
+			require.NoError(t, err)
+			// give enough time to config manager to reload the previously written config
+			time.Sleep(writeDelay)
```

Comment:
If possible, it is better to avoid sleeping in tests and instead of that use polling for success conditions, like:
```go
	require.EventuallyWithT(t, func(t *assert.CollectT) {
		assert.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(fmt.Sprintf(`
                # HELP weaviate_runtime_config_hash Hash value of the currently active runtime configuration
        	# TYPE weaviate_runtime_config_hash gauge
        	weaviate_runtime_config_hash{sha256="%s"} 1
		# HELP weaviate_runtime_config_last_load_success Whether the last loading attempt of runtime config was success
		# TYPE weaviate_runtime_config_last_load_success gauge
		weaviate_runtime_config_last_load_success 1
		`, fmt.Sprintf("%x", sha256.Sum256(buf))))))
	}, time.Second, 10*time.Millisecond)
```

Reply:
good point. I didn't know about `require.EventuallyWithT` helper :) 

In fact[ I wrote separate `Poll` helper for this purpose](https://github.com/weaviate/weaviate/blob/4b97f10e3d47bfb99753383485f8d9f569effe35/usecases/config/runtime/manager_test.go#L393-L414) then removed it to avoid any complications. But this helper solves my problem. Thanks :)